plugins/hls-tactics-plugin/src/Wingman/AbstractLSP/Types.hs

```diff
@@ -121,19 +121,13 @@ data Continuation sort target payload = Continuation
 -- | What file are we looking at, and what bit of it?
 data FileContext = FileContext
   { fc_uri      :: Uri
-  , fc_nfp      :: NormalizedFilePath
   , fc_range    :: Maybe (Tracked 'Current Range)
     -- ^ For code actions, this is 'Just'. For code lenses, you'll get
     -- a 'Nothing' in the request, and a 'Just' in the response.
   }
   deriving stock (Eq, Ord, Show, Generic)
   deriving anyclass (A.ToJSON, A.FromJSON)
 
-deriving anyclass instance A.ToJSON NormalizedFilePath
```

Comment:
The problem is pretty obvious when you look at these :)

Reply:
Maybe there should be an `instance TypeError "Do not do this" => ToJSON NormalizedFilePath` definition in the `lsp` package where `NormalizedFilePath` is defined, to prevent this weird bug from happening again

Reply:
I don't remember anymore. Presumably the derived instances are not roundtripping, i.e. `fromJSON . receiveLSP . sendLSP . toJSON /= id`, because of the embedded hashes.